src/integration_fleets.md

```diff
@@ -31,11 +34,53 @@ The basic workflow of developing a fleet adapter is the following:
+4. Fill in the missing code in [`RobotClientAPI.py`](https://github.com/open-rmf/fleet_adapter_template/blob/main/fleet_adapter_template/fleet_adapter_template/RobotClientAPI.py). This is the most important part of integration. Depending on the API you choose to interface between the adapter and manager, you will have to format your robot's data accordingly and return the values specified in the template. This is critical as the functions in `RobotClientAPI` are called from `RobotCommandHandle` when the adapter is planning tasks or updating traffic schedules.
+
+Once you are done, you can run both the fleet adapter and your custom fleet manager. Remember to parse the configuration file and navigation graph when launching the adapter.
+
+## Case Study: RMF Demos Fleet Adapter
+
+The Python implementation of the **Full Control** fleet adapter classes is demonstrated in the [demos fleet adapter](https://github.com/xiyuoh/rmf_demos/tree/feature/demo_fleet_adapter/rmf_demos_fleet_adapter). Building on top of the fleet adapter template, the demos fleet adapter uses REST API as an interface between the adapter and the simulation robots: the adapter sends out commands to the robots, while the robots update the adapter on their current state information. This is done by creating a [`fleet_manager`](https://github.com/xiyuoh/rmf_demos/blob/feature/demo_fleet_adapter/rmf_demos_fleet_adapter/rmf_demos_fleet_adapter/fleet_manager.py) node that contains the necessary REST endpoints for `RobotClientAPI` to interact with.
+
+<img src="images/demo_fleet_adapter_flow.png">
+
+### Demos Fleet Manager
+
+Whenever a command is ready to be sent from the adapter's `RobotCommandHandle`, it will call the relevant API function defined in `RobotClientAPI` and query the corresponding endpoint from the API server in the `fleet_manager` node. Each function either retrieves specific information about the robot's current state (including but not limited to its last known position, remaining battery level, and whether it has completed a request), or sends a command to the robot to carry out a request. The robot state information is required for the fleet adapter to update the traffic schedule, plan for subsequent tasks, and guide robots across different paths in the environment.
+
+The demos fleet adapter is integrated with the simulation robots which publish their state information via internal ROS2 messages, hence the `fleet_manager` also serves to consolidate the messages published by different robots from its fleet and sends them to the right robot's `RobotCommandHandle`. The API endpoints are designed such that the adapter can query information or send commands to a particular robot by specifying the robot's name. The `fleet_manager` will ensure that the robot exists in the fleet before returning the requested state information or relaying commands to the simulation robot. Additionally, the adapter can retrieve the status of the entire fleet's robots.
+
+### FastAPI
```

Comment:
Can leave out this portion of `FastAPI` as its an implementation detail of the demo fleet manager that we have created and not relevant to users who will look at this document to understand how to integrate their fleet with RMF. Maybe we can add this to the readme in rmf_demos_fleet_adapter instead

Reply:
Got it :+1: already in rmf_demos_fleet_adapter README